iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp

```diff
@@ -57,7 +57,8 @@ class RouDi
           const config::MonitoringMode f_monitoringMode = config::MonitoringMode::ON,
           const bool f_killProcessesInDestructor = true,
           const MQThreadStart mqThreadStart = MQThreadStart::IMMEDIATE,
-          const version::CompatibilityCheckLevel compatibilityCheckLevel = version::CompatibilityCheckLevel::PATCH);
+          const version::CompatibilityCheckLevel compatibilityCheckLevel = version::CompatibilityCheckLevel::PATCH,
```

Comment:
I think we reached now the maximum number of arguments for the good old roudi ctor. I would suggest the following:
1. Create a struct called `struct RoudiStartupParameters` and move all the arguments from here in there.
2. Use this struct `RoudiStartupParameters` in this ctor instead of all the arguments which provides us with the big advantage that we only have to change one place if we add another argument to roudi.
3. Use it like
```cpp
RouDi myRoudi(RoudiStartupParameters{memoryInterface, portManager, config::MonitoringMode::ON ....});
```
then we do not have to change much in the source code. We just have to add `RoudiStartupParameters`

Reply:
What about creating a new issue for that with a separate PR to follow?! That'll help keep to keep our PRs smaller.

Reply:
In this case I would disagree. The code changes should be minor and the struct would only need some member without any logic at all.

Reply:
we already have something which we could use, it's the `RouDiConfig` ;)

Reply:
Checkout latest commit.